src/domain/synthetics/positions/utils.ts

```diff
@@ -245,3 +246,47 @@ export function formatLeverage(leverage?: BigNumber) {
 
   return `${formatAmount(leverage, 4, 2)}x`;
 }
+
+export function getEstimatedLiquidationTime(
+  position: PositionInfo,
+  minCollateralUsd: BigNumber | undefined,
+  period = "1h"
+): number {
+  const { marketInfo, isLong, sizeInUsd, isOpening, netValue } = position;
+
+  if (isOpening || !minCollateralUsd) return 0;
+
+  let liquidationCollateralUsd = applyFactor(sizeInUsd, marketInfo.minCollateralFactor);
+  if (liquidationCollateralUsd.lt(minCollateralUsd)) {
+    liquidationCollateralUsd = minCollateralUsd;
+  }
+  const borrowFeePerHour = getBorrowingFeeRateUsd(marketInfo, isLong, sizeInUsd, CHART_PERIODS[period]);
+  const fundingFeePerHour = getFundingFeeRateUsd(marketInfo, isLong, sizeInUsd, CHART_PERIODS[period]);
+  const priceImpactDeltaUsd = getPriceImpactForPosition(marketInfo, sizeInUsd.mul(-1), isLong, {
```

Comment:
we also need limit price impact with maxNegativePriceImpactForLiquidations

The same as in getLiquidationPrice, maybe we can create function line "getPriceImpactForLiquidation" to avoid duplication
```
 const maxNegativePriceImpactUsd = applyFactor(sizeInUsd, marketInfo.maxPositionImpactFactorForLiquidations).mul(-1);

  if (priceImpactDeltaUsd.lt(maxNegativePriceImpactUsd)) {
      priceImpactDeltaUsd = maxNegativePriceImpactUsd;
  }

 // Ignore positive price impact
 if (priceImpactDeltaUsd.gt(0)) {
      priceImpactDeltaUsd = BigNumber.from(0);
 }
```

Reply:
Fixed!